### PR TITLE
Add a Disk.getTransformStream method.

### PR DESCRIPTION
**Pull request by zvin**
This method returns a stream.Transform that adds the disk recorded
writes to the original disk stream.

Connects-To: #8

Change-Type: minor
---- Autogenerated Waffleboard Connection: Connects to #8

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -42,6 +42,52 @@ export class DiskStream extends Readable {
...
+	public _transform(chunk: Buffer, enc: string, cb: () => void): void {
+		const start: number = this.position;
+		const end: number = start + chunk.length - 1;
+		const interval: Interval = [ start, end ];
+		while (this.currentChunk) {
+			if (intervalIntersection(this.currentChunk.interval(), interval)) {
+				const buf = this.currentChunk.data();
+				const startShift = this.currentChunk.start - start;
+				const endShift = this.currentChunk.end - end;
+				const sourceStart = -Math.min(startShift, 0);
```
Comment (jhermsmeier) [inline src/index.ts]: Why a negative `sourceStart` here – isn't that going to be OOB?

Comment (zvin) [reply src/index.ts]: @jhermsmeier 
`sourceStart` can not be negative here:

 * if `startShift` is negative, `sourceStart` will be `- startShift` : positive
 * if `startShift` is positive, `sourceStart` will be 0

This is equivalent to:
```typescript
const sourceStart = (startShift < 0) ? -startShift : 0
```

File: src/index.ts [hunk 2]
```diff
@@ -1,6 +1,6 @@
 import * as AWS from 'aws-sdk';
 import * as Bluebird from 'bluebird';
-import { Readable } from 'stream';
+import { Readable, Transform } from 'stream';
```
[the remark this answers is not in the document]
Comment (zvin) [reply src/index.ts]: @jhermsmeier this project only supports node 6 and above, so I don't think this will be useful.

File: src/index.ts (hunk 1 above)
Comment (jhermsmeier) [reply src/index.ts]: I see; I think `const sourceStart = (startShift < 0) ? -startShift : 0` is a bit more clear to the reader

File: src/index.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (jhermsmeier) [reply src/index.ts]: There are some things, like i.e. `.destroy()` that were only added in Node v8 though – which won't work on streams by default in Node v6 and lower

[the remark this answers is not in the document]
Comment (zvin) [reply src/index.ts]: @jhermsmeier 
We don't use these new methods (`_destroy` or `_final`) and `readable-stream` does not have a `@types` package for typescript so for now, I'll stick to node's `Stream` and `Transform`.

But I agree that would be a good thing to have in the future.

File: src/index.ts (hunk 1 above)
Comment (zvin) [reply src/index.ts]: That's correct, updated.

**Pull request merged**